File: app/src/main/java/com/zighter/zighterandroid/util/IconHelper.java
package com.zighter.zighterandroid.util;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;

import com.mapbox.mapboxsdk.annotations.Icon;
import com.mapbox.mapboxsdk.annotations.IconFactory;
import com.zighter.zighterandroid.R;

public class IconHelper {

    @NonNull
    public static Icon getIcon(@NonNull Context context, @NonNull Type type) {
        int drawableRes;
        switch (type) {
            case SIGHT:
                drawableRes = R.drawable.ic_navigation_black;
                break;
            case ENDPOINT:
                drawableRes = R.drawable.ic_sight_black;
                break;
            case CHECKED_SIGHT:
                drawableRes = R.drawable.ic_checked_sight;
                break;
            default: throw new IllegalStateException("Unknown icon type");
        }

        return IconFactory.getInstance(context).fromResource(drawableRes);
    }

    public enum Type {
        ENDPOINT,
        SIGHT,
        CHECKED_SIGHT
    }

}

File: app/src/main/java/com/zighter/zighterandroid/util/IconHelper.kt
@file:JvmName("IconHelper")

package com.zighter.zighterandroid.util

import android.content.Context

import com.mapbox.mapboxsdk.annotations.Icon
import com.mapbox.mapboxsdk.annotations.IconFactory
import com.zighter.zighterandroid.R


fun getIcon(context: Context, type: IconHelperType): Icon {
    val drawableRes: Int = when (type) {
        IconHelperType.SIGHT -> R.drawable.ic_navigation_black
        IconHelperType.ENDPOINT -> R.drawable.ic_sight_black
        IconHelperType.CHECKED_SIGHT -> R.drawable.ic_checked_sight
        IconHelperType.CURRENT_LOCATION -> R.mipmap.ic_accessibility
        IconHelperType.CURRENT_LOCATION_OFFLINE -> R.mipmap.ic_accessibility_offline
        else -> throw IllegalStateException("Unknown icon type")
    }

    return IconFactory.getInstance(context).fromResource(drawableRes)
}

enum class IconHelperType {
    ENDPOINT,
    SIGHT,
    CHECKED_SIGHT,
    CURRENT_LOCATION,
    CURRENT_LOCATION_OFFLINE
}

